src/coreclr/jit/emitriscv64.cpp

```diff
@@ -3503,51 +3510,96 @@ void emitter::emitDispInsName(
+                    }
+                    printLength = printf("slli");
+                    imm12 &= 0x3f; // 6 BITS for SHAMT in RISCV64
+                }
+                break;
                 case 0x2: // SLTI
-                    printf("slti           %s, %s, %d\n", rd, rs1, imm12);
-                    return;
+                    printLength = printf("slti");
+                    break;
                 case 0x3: // SLTIU
-                    printf("sltiu          %s, %s, %d\n", rd, rs1, imm12);
-                    return;
+                    printLength = printf("sltiu");
+                    break;
                 case 0x4: // XORI
-                    printf("xori           %s, %s, 0x%x\n", rd, rs1, imm12);
-                    return;
+                    printLength = printf("xori");
+                    isHex       = true;
+                    break;
                 case 0x5: // SRLI & SRAI
-                    if (((code >> 30) & 0x1) == 0)
-                    {
-                        printf("srli           %s, %s, %d\n", rd, rs1, imm12 & 0x3f); // 6BITS for SHAMT in RISCV64
-                    }
-                    else
+                {
+                    static constexpr unsigned kLogicalShiftFunct6    = 0b000000;
+                    static constexpr unsigned kArithmeticShiftFunct6 = 0b010000;
+
+                    unsigned funct6         = (imm12 >> 6) & 0x3f;
+                    bool     isLogicalShift = funct6 == kLogicalShiftFunct6;
+                    if ((!isLogicalShift) && (funct6 != kArithmeticShiftFunct6))
                     {
-                        printf("srai           %s, %s, %d\n", rd, rs1, imm12 & 0x3f); // 6BITS for SHAMT in RISCV64
+                        return emitDispIllegalInstruction(code);
                     }
-                    return;
+                    printLength = printf(isLogicalShift ? "srli" : "srai");
+                    imm12 &= 0x3f; // 6BITS for SHAMT in RISCV64
+                }
+                break;
                 case 0x6: // ORI
-                    printf("ori            %s, %s, 0x%x\n", rd, rs1, imm12 & 0xfff);
-                    return;
+                    printLength = printf("ori");
+                    imm12 &= 0xfff;
+                    isHex = true;
+                    break;
                 case 0x7: // ANDI
-                    printf("andi           %s, %s, 0x%x\n", rd, rs1, imm12 & 0xfff);
-                    return;
+                    printLength = printf("andi");
+                    imm12 &= 0xfff;
+                    isHex = true;
+                    break;
                 default:
-                    printf("RISCV64 illegal instruction: 0x%08X\n", code);
-                    return;
+                    return emitDispIllegalInstruction(code);
             }
+            assert(printLength > 0);
+            int paddingLength = kMaxInstructionLength - printLength;
+            assert(paddingLength > 0);
```

Comment:
Note: technicallly 0 padding is also fine, but since you're not handling long names here like amomaxu.w.aqrl, it will never fire.

Reply:
Ok, removed assert